test-common/src/main/java/de/bwaldvogel/mongo/backend/AbstractOplogTest.java

```diff
@@ -456,4 +463,67 @@ public void testOplogShouldFilterNamespaceOnChangeStreams() throws Exception {
         return subscriber.values().get(0);
     }
 
+    @Test
+    @Disabled
```

Comment:
Why is the test `@Disabled`? The test does not fail for me.

Actually I’m not able to grasp what this test is trying to test/show.
The intensive use of `RxJava` doesn’t necessarily help to understand the test. I’m not even sure what happens if the test breaks in the middle? Which code takes care of cleaning up potentially remaining subscriptions?

If I understood the basic idea correctly, it should be as simple as starting one or two threads that subscribe a change stream and then insert documents in the test "main" thread.
The thread sleeps can then be usually avoided for example by using a `CyclicBarrier` to make the test fully deterministic.
However, the test should somehow explain/show where the 100% CPU usage happens…

Reply:
the test is enabled in a later commit that contains the fix.

you have understood the idea of the test correctly. we start a number of change streams, insert a number of documents, and then assert that all of the watches saw the same items emitted within a sensible timeout. 

we are maintaining an internal fork of the library. without the fix to introduce a delay, we are finding that some of our unit tests fail. the problem is more acute as number of change streams increases and especially on resource constrained hardware e.g. busy CI servers. I will take another pass at this test to ensure it always fails without the fix. 

I can see high CPU usage when running the test without the fix and see it drop significantly when the delay is added.

the tear down is handled by the Rx Test Subscriber which cancels all the subscriptions. I'll refactor to remove the Thread sleep.